bodhi/server/notifications.py

```diff
@@ -97,7 +98,68 @@ def send_fedmsgs_after_commit(session):
             session.info['fedmsg'][topic] = []
 
 
+@event.listens_for(Session, 'after_commit')
+def send_messages_after_commit(session):
+    """
+    Send messages via AMQP after a database commit occurs.
+
+    Args:
+        session (sqlalchemy.orm.session.Session): The session that was committed.
+    """
+    if 'messages' in session.info:
+        for m in session.info['messages']:
+            try:
+                api.publish(m)
+            except fml_exceptions.BaseException:
+                # In the future we should handle errors more gracefully
+                _log.exception("An error occurred publishing %r after a database commit", m)
+        session.info['messages'] = []
+
+
 def publish(topic, msg, force=False):
+    """
+    Send a message via AMQP or fedmsg.
+
+    This use used to send a message to fedmsg if "fedmsg_enabled" is true in the
+    application configuration, otherwise it will use AMQP.
+
+    Args:
+        topic (str): The topic suffix. The "bodhi" prefix is applied, along with
+            the "topic_prefix" and "environment" settings from fedmsg.
+        msg (dict): The message body to send.
+        force (bool): If False (the default), the message is only sent after the
+            currently active database transaction successfully commits. If true,
+            the messages is sent immediately.
+    """
+    if bodhi.server.config.config.get('fedmsg_enabled'):
+        _fedmsg_publish(topic, msg, force)
+        return
+
+    # Dirty, nasty hack that I feel shame for: fedmsg modifies messages quietly
+    # if they have objects with __json__ methods on them. For now, copy that
+    # behavior. In the future, callers should pass fedora_messaging.api.Message
+    # sub-classes or this whole API should go away.
+    body = fedmsg.encoding.loads(fedmsg.encoding.dumps(msg))
+
+    # Dirty, nasty hack #2: fedmsg mangles the topic in various ways. Duplicate
+    # that behavior here for now. In the future callers should pass proper topics.
+    fedmsg_conf = fedmsg.config.load_config()
+    topic = "{prefix}.{env}.bodhi.{t}".format(
+        prefix=fedmsg_conf["topic_prefix"], env=fedmsg_conf["environment"], t=topic)
```

Comment:
It would be very nice to fix the topics, but I think maybe the best migration path is something like:

1. Add new message types with reasonable topics while still publishing old messages with old topics.
2. Publish both old and new messages.
3. Make an announcement about the old messages going away, convert all known consumers of those messages to the new message
4. Drop the old messages.

There's not really anything in fedora-messaging to help you in this, although if we find a common pattern that most apps would use we should definitely add that.